#!/bin/bash

#PBS -q normal
#PBS -l select=1:ncpus=16:mem=64GB
#PBS -l walltime=24:00:00
#PBS -P Personal 
#PBS -N STAR3
#PBS -o /home/users/ntu/muhamma4/output/STAR3.o
#PBS -e /home/users/ntu/muhamma4/output/STAR3.e

## Liver and Heart First
EXPT=Human

THREADS=16

rawPath='/home/users/ntu/muhamma4/scratch/raw_data/'
outPath='/home/users/ntu/muhamma4/scratch/STAR_result_2/'
GENOME_DIR='/home/users/ntu/muhamma4/scratch/reference_STAR/Homo_sapiens/NCBI/GRCh38Decoy/STAR/'
GTF_FILE='/home/users/ntu/muhamma4/scratch/reference_STAR/Homo_sapiens/NCBI/GRCh38Decoy/Annotation/Genes.gencode/genes.gtf'

RESULTS=$outPath/RESULTS_FEATURECOUNT_${EXPT}

function process_STAR {
	PREFIX=$3
	echo $(date +"%Y-%m-%d %H:%M:%S") " - STAR PROCESSING ${PREFIX}"
	STAR --genomeDir $GENOME_DIR --sjdbGTFfile $GTF_FILE --readFilesIn $1 $2 --runThreadN $THREADS --twopassMode Basic --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --outFileNamePrefix $3 1>$3-STAR.o 2>$3-STAR.e
	
	mv ${PREFIX}Aligned.sortedByCoord.out.bam $RESULTS/${PREFIX}.out.bam
	echo $(date +"%Y-%m-%d %H:%M:%S") " - STAR COMPLETED ${PREFIX}"
}

function create_folder {
  	if [ -d $1 ]; then
    	rm -r $1
  	fi
  	mkdir $1
}

module load anaconda/3
source activate /home/users/ntu/muhamma4/.conda/envs/BS6214-2

x='_1.fastq.gz'
y='_2.fastq.gz'
# create_folder $outPath
# create_folder $RESULTS
cat /home/users/ntu/muhamma4/scripts/SRA_list.txt | while read p
do
	cd $outPath
	create_folder $p
	cd $p

	process_STAR $rawPath$p$x $rawPath$p$y $p
done

source deactivate
module unload anaconda/3
echo All Done
